### PR TITLE
Add E2E test for script link and mobile menu

### PR DESCRIPTION
**Pull request by tochwill**
Resolves #NUMBER

**Overall change:** 
_A very high-level summary of easily-reproducible changes that can be understood by non-devs._

**Code changes:**
- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: cypress/integration/pages/testsForAllPages.js [hunk 1]
```diff
@@ -318,6 +318,14 @@ export const testsThatFollowSmokeTestConfigforAllPages = ({
...
       });
 
+      if (variant !== 'default') {
+        it('should render a script switch component, if not the default variant', () => {
+          cy.get('header')
+            .find('a[data-variant]')
+            .should('have.lengthOf', 1);
+        });
+      }
+
```
Comment (PriyaKR) [inline cypress/integration/pages/testsForAllPages.js:334]: Can we add a line of code to assert whether the script switcher is working? or thats something covered by the unit tests?

File: cypress/integration/pages/testsForAllPages.js [hunk 2]
```diff
@@ -341,6 +349,34 @@ export const testsThatFollowSmokeTestConfigforAllPages = ({
...
+                .should('not.be.visible');
+
+              cy.get('nav')
+                .find('ul[class^="DropdownUl"]')
+                .should('be.visible');
+            });
+          }
         }
       }
     });
```
Comment (PriyaKR) [inline cypress/integration/pages/testsForAllPages.js:388]: The tests for navigation and the dropdown looks good to me.Wondering whether the individual links in the nav dropdown are covered by unit tests.

File: cypress/integration/pages/testsForAllPages.js (hunk 1 above)
Comment (tochwill) [reply cypress/integration/pages/testsForAllPages.js:334]: The unit test that I added as part of this PR makes sure that the link has the correct `href` based on the variant provided - do you think a test that clicks the link and ensures that the variant page loads is a good idea?

File: cypress/integration/pages/testsForAllPages.js (hunk 2 above)
Comment (tochwill) [reply cypress/integration/pages/testsForAllPages.js:388]: We currently have unit tests to ensure that the lists passed to the navigation component are rendered, and snapshot tests covering that the navigation items are rendered for a particular service

Comment (PriyaKR) [reply cypress/integration/pages/testsForAllPages.js:388]: Do we have tests to assert the href for the nav items?

File: cypress/integration/pages/testsForAllPages.js (hunk 1 above)
Comment (PriyaKR) [reply cypress/integration/pages/testsForAllPages.js:334]: i guess as long as we have assertion for href that should be enough

File: cypress/integration/pages/testsForAllPages.js (hunk 2 above)
Comment (tochwill) [reply cypress/integration/pages/testsForAllPages.js:388]: I'll add an assertion unit test in to ensure that the links are correctly formed :)

**Review by AlistairGempf — APPROVED** [review]
Nice, good separation of things to be unit tested and integration tested 👍

**Comment by Bopchy** [comment]
Looks good. Just have a question on the changes in https://github.com/bbc/simorgh/pull/5157/files#diff-fac8f4d303472ce8af320474fa6d35a0. Wondering why the tests were moved to `cypress/integration/pages/testsForAllPages.js` ?

**Review by Bopchy — APPROVED** [review]

File: cypress/integration/pages/testsForAllPages.js [hunk 3]
```diff
@@ -318,6 +318,14 @@ export const testsThatFollowSmokeTestConfigforAllPages = ({
           .should('have.attr', 'href', '#content');
       });
 
+      if (variant !== 'default') {
```
Comment (tochwill) [inline cypress/integration/pages/testsForAllPages.js]: I think I'll check that there _isn't_ a script switch when there shouldn't be as well

**Comment by tochwill** [comment]
> Looks good. Just have a question on the changes in https://github.com/bbc/simorgh/pull/5157/files#diff-fac8f4d303472ce8af320474fa6d35a0. Wondering why the tests were moved to `cypress/integration/pages/testsForAllPages.js` ?

Hey! We had nothing covering that the mobile menu was rendered, so thought that it would be useful to run these for canonical as well.

**Pull request merged**